### PR TITLE
"CAMEL:19341:camel-spring-boot - Example with ActiveMQ Artemis"

### PR DESCRIPTION
**Pull request by ramu11**
camel-spring-boot - Example with ActiveMQ Artemis using Artemis client library

### CODE REVIEW

File: artemis/pom.xml [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+            <artifactId>camel-spring-boot-starter</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.activemq</groupId>
+            <artifactId>artemis-jms-client</artifactId>
+            <version>${artemis-jms-client-version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.camel</groupId>
+            <artifactId>camel-jms</artifactId>
```
Comment (davsclaus) [inline artemis/pom.xml:88]: you do not need this as you should only use -starter for Camel JARs

File: artemis/src/main/data/.camel/ReadMe.txt [hunk 2]
```diff
@@ -0,0 +1,43 @@
+example project which connects to A-MQ 7 from Fuse 7, using remote A-MQ address
```
Comment (davsclaus) [inline artemis/src/main/data/.camel/ReadMe.txt:1]: Maybe this .camel file was added by mistake

File: artemis/src/main/java/sample/camel/JmsConfig.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    public String getRemoteUri() {
+        return remoteUri;
+    }
+    
+    public void setRemoteUri(String remoteUri) {
+        this.remoteUri = remoteUri;
+    }
+
+    @Bean
+    public org.apache.activemq.artemis.jms.client.ActiveMQJMSConnectionFactory artemisConnectionFactory() throws JMSException{
```
Comment (davsclaus) [inline artemis/src/main/java/sample/camel/JmsConfig.java:80]: Can we not do this better? Why should end users write a java class to configure Artemis? This is bad. I assume SB has a -starter auto configuration we can use or something.

File: artemis/src/main/java/sample/camel/SampleAutowiredAmqpRoute.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.stereotype.Component;
+
+@Component
+public class SampleAutowiredAmqpRoute extends RouteBuilder {
+
+    @Autowired ActiveMQJMSConnectionFactory artemisConnectionFactory;
+    @Bean
+    public org.apache.camel.component.jms.JmsComponent jms() {
```
Comment (davsclaus) [inline artemis/src/main/java/sample/camel/SampleAutowiredAmqpRoute.java:30]: We should find a way to make this not needed, eg a SB artemis starter or something.

**Review by davsclaus — CHANGES_REQUESTED** [review]

**Comment by davsclaus** [comment]
You can configure SB something like
https://docs.spring.io/spring-boot/docs/2.0.x/reference/html/boot-features-messaging.html#boot-features-artemis

**Comment by ramu11** [comment]
This pr got messed up. closing this to create new one

**Pull request closed without merging**